jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoost.scala

```diff
@@ -123,7 +131,10 @@ object XGBoost extends Serializable {
           }
           val partitionItr = fromDenseToSparseLabeledPoints(trainingSamples, missing)
           val trainingSet = new DMatrix(new JDMatrix(partitionItr, cacheFileName))
-          booster = SXGBoost.train(trainingSet, xgBoostConfMap, round,
+          if (groupData != null) {
+            trainingSet.setGroup(groupData(partIndex).toArray)
+          }
```

Comment:
say 'xgBoostConfMap' has M bytes,  'groupData' has N bytes, with a naive (and incorrect) xgBoostConfMapFiltered will have M - N bytes (which should be larger than M-N considering JVM overhead, etc.)

You still call groupData(partIndex) in the closure of mapPartition, so groupData (N bytes) will still be serialized and pass to executors, and xgBoostConfMapFiltered is also called within mapPartition, so another (M - N) will be serialized and passed to executors...

I didn't see any difference here